### PR TITLE
[IMP] web_planner: Remove foreign constraints

### PR DESCRIPTION
**Pull request by pedrobaeza**
Remove foreign constraints on merged `web_planner` module for avoiding errors when removing referenced records.

@Tecnativa

### CODE REVIEW

**Review by cubells — APPROVED** [review]
:+1: 

LGTM

Code review and test

File: addons/web/migrations/12.0.1.0/pre-migration.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+from psycopg2.extensions import AsIs
+from openupgradelib import openupgrade
+
+
+def remove_web_planner_constraints(env):
+    """Remove foreign constraints on merged `web_planner` module for avoiding
+    errors when removing referenced records.
+    """
+    constraint_list =  [
```
Comment (MiquelRForgeFlow) [inline addons/web/migrations/12.0.1.0/pre-migration.py]: flake8

Comment (pedrobaeza) [reply addons/web/migrations/12.0.1.0/pre-migration.py]: Fixed. Please squash on merge. I remember someone talking about this problem in one PR, but I'm not able to find the thread.

File: addons/web/migrations/12.0.1.0/pre-migration.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    constraint_list = [
+        'web_planner_create_uid_fkey',
+        'web_planner_menu_id_fkey',
+        'web_planner_view_id_fkey',
+        'web_planner_write_uid_fkey',
+    ]
+    for constraint in constraint_list:
+        openupgrade.logged_query(
+            env.cr, "ALTER TABLE web_planner DROP CONSTRAINT %s",
+            (AsIs(constraint))
```
Comment (MiquelRForgeFlow) [inline addons/web/migrations/12.0.1.0/pre-migration.py]: `(AsIs(constraint), ),`

ValueError("SQL query parameters should be a tuple, list or dict; got %r" % (params,))

Comment (pedrobaeza) [reply addons/web/migrations/12.0.1.0/pre-migration.py]: Solved

**Comment by MiquelRForgeFlow** [comment]
@pedrobaeza I don't understand. Why it gives problems in travis-ci?

**Review by pedroslopez — APPROVED** [review]
Wonderful! There were multiple things that lingered around because of this.

**Comment by pedrobaeza** [comment]
Travis is flickering too much...

**Review by MiquelRForgeFlow — APPROVED** [review]

**Pull request merged**